server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -686,13 +689,41 @@ public String toString() {
             return "tsid";
         }
 
+        /**
+         * @param value The TSID as a {@link BytesRef}
+         * @return the Base 64 encoded TSID
+         */
         @Override
         public Object format(BytesRef value) {
-            return TimeSeriesIdFieldMapper.decodeTsid(new BytesArray(value).streamInput());
+            try {
+                // NOTE: if the tsid is a map of dimension key/value pairs (as it was before introducing
+                // tsid hashing) we just decode the map and return it.
+                return TimeSeriesIdFieldMapper.decodeTsidAsMap(value);
+            } catch (Throwable t) {
```

Comment:
I think we should catch `Exception` instead here.

Reply:
Using `Throwable` because `UnicodeUtil#UTF8toUTF16` called by `BytesRef#utf8ToString` uses assertions when trying to decode a String. Anyway this is an issue since assertions are removed in release builds...so I expect this to fail with an `AssertionError` only when testing in debug mode (in non-release builds). I expect this to throw some other Exception in release builds.

Reply:
Any exception that extends from `Error` should get caught by `ElasticsearchUncaughtExceptionHandler`. So if a catch clause catches `Throwable` we're at risk of hiding errors where it is better to halt the process. This also includes `AssertionError` (in general it could leave things in an inconsistent state), in production we never run with assertions enabled, but for integration tests this kill the test cluster quickly and logs the stacktrace. Assertions are meant for (integration) tests only. I think this is the reason we never catch `Throwable` in production code.

Reply:
Btw I'm ok with just catching AssertionError the way you did here - the above was more a general comment.